e2etest/config.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package e2etest
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/JeffreyRichter/enum/enum"
+)
+
+// clearly define all the inputs to the end-to-end tests
+// it's ok to panic if the inputs are absolutely required
+// the general guidance is to take in as few parameters as possible
+type GlobalInputManager struct{}
+
+func (GlobalInputManager) GetAccountAndKey(accountType AccountType) (string, string) {
```

Comment:
How do we test OAuth, if we start with account keys?  If that's not possible, should we abandon account keys, and go with OAuth tokens as our single type of Azure Auth. (Probably with Serivce Principal auth, I'm guesssing.  Might be able to give that Principal rights to make accounts even, which would help with account-to-accountt scenarios).  Then we'd need just one sent of SPA creds (e.g. a cert) and the account/key for S3

Reply:
Yes, that's what I had in mind too. SPN with access to the entire subscription should be the easiest approach to do account -> account tests.

However, it's somewhat expensive (long) to create new accounts, so for regular tests, I lean towards using an account like we do now. Let's discuss offline.

Reply:
I think a secret supplied from an environment variable would work just fine. As long as the service principal has the ability to create user delegation keys and such, it'll work just fine.

---
In reply to: [370330965](https://github.com/Azure/azure-storage-azcopy/pull/842#discussion_r370330965) [](ancestors = 370330965)